### ISSUE TITLE
Remove distracted red bar from IDLE for some SyntaxErrors

### ISSUE THREAD

**Issue by aroberge** [python/cpython#93966]
**Bug report**

This may be more an distracting cosmetic issue than a true bug, but it is something that can be fixed easily.

As Raymond Hettinger mentioned a while ago on Twitter (https://twitter.com/raymondh/status/1501700903870468099), when a SyntaxError is raised and the location corresponds to the "\n" at the end of a line, a huge red highlighted region is shown at the end of the invalid line.

For example, type
```
from math
```
followed by enter. This red highlighted line will remain on the screen until another SyntaxError occurs.
Unlike cases where a SyntaxError highlights just a few characters pointing at a specific problem, this adds really no informative value, once the user starts entering some new code.

**Your environment**

I have reproduced this will IDLE on Windows 10 on all Python versions I have installed, from 3.6 to 3.11 inclusively.

**Proposed fix**

I have been able to fix this on a custom version of IDLE (friendly_idle) by doing the following:

In `pyshell.ModifiedInterpreter.transfer_path` which sets up some values at the beginning of a new process,
define the following:
```python
self.error_at_end_of_line = False
```

In `pyshell.ModifiedInterpreter.showsyntaxerror`, add the last line shown below:
```python
        if lineno == 1:
            pos = "iomark + %d chars" % (offset - 1)
        else:
            pos = "iomark linestart + %d lines + %d chars" % (lineno - 1, offset - 1)
        self.error_at_end_of_line = text.get(pos) == "\n"  # <-- add this
```
At the beginning of `pyshell.ModifiedInterpreter.runsource`, add the following:
```python
        if self.error_at_end_of_line:
            self.error_at_end_of_line = False
            self.tkconsole.text.tag_remove("ERROR", "1.0", "end")
```

If desired, I could do a pull request that would include these changes.

### Linked PRs
* gh-157595
* gh-158057
* gh-158058
* gh-158059

**Comment by terryjreedy** [comment]
The bar is what tk does when one tags '\n'.  I don't know of anyway to tag just one space.

In the editor, any edit to the line with the SyntaxError tag dismisses the tag.  In Shell, I think waiting until one submits a new statement is fine, but I see no reason not to then dismiss any such tag.   So adding the new flag is not needed and the only addition needed is the tag_ remove() without the condition.

I believe that adding the line to def runsource() should make the existing tag_remove in def showsyntaxerror.  Try commenting out the latter and testing with a repeat of the syntax error to make sure before removing it.

**Comment by aroberge** [comment]
Yes, simply adding `self.tkconsole.text.tag_remove("ERROR", "1.0", "end")` at the beginning of `runsource` is sufficient. Then, `text.tag_remove("ERROR", "1.0", "end")` can be removed from `showsyntaxerror` as it becomes superfluous.   

I did not suggest this simpler change as I thought that the original thinking was there was some value in keeping the highlighting when there is only one, or perhaps a few characters highlighted.

**Comment by serhiy-storchaka** [comment]
#157595 does what was agreed above: the highlighting is removed when the next statement is entered.
